### ISSUE TITLE
R packages not found by histoneHMM in ChIP-Seq pipeline

### ISSUE THREAD

**Issue by Remi-Montagne** [maxplanck-ie/snakepipes#451]
Hello,

I am trying to run the ChIPSeq pipeline but it looks like there are problem when running it and looking for broad peaks with histoneHMM.

Here is command line: 
``` bash
ChIP-seq -d Outputs/ sacCer3R64 Outputs/chip_seq_sample_config.PREDICTED.yaml --local --single-end --DAG
```

And here is the error message that I get:
``` bash
Attaching package: ‘histoneHMM’

The following object is masked from ‘package:base’:

    truncate

Loading required package: stats4
Loading required package: BiocGenerics
Loading required package: parallel

Attaching package: ‘BiocGenerics’

The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB

The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs

The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, basename, cbind, colMeans,
    colnames, colSums, dirname, do.call, duplicated, eval, evalq,
    Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply,
    lengths, Map, mapply, match, mget, order, paste, pmax, pmax.int,
    pmin, pmin.int, Position, rank, rbind, Reduce, rowMeans, rownames,
    rowSums, sapply, setdiff, sort, table, tapply, union, unique,
    unsplit, which, which.max, which.min

Loading required package: S4Vectors

Attaching package: ‘S4Vectors’

The following object is masked from ‘package:base’:

    expand.grid

Loading required package: IRanges
Loading required package: GenomeInfoDb
Loading required package: MASS
Loading required package: Rsamtools
Loading required package: Biostrings
Loading required package: XVector

Attaching package: ‘Biostrings’

The following object is masked from ‘package:base’:

    strsplit

There were 30 warnings (use warnings() to see them)
Error in dn[[2L]] : subscript out of bounds
Calls: GR2gff ... .local -> lapply -> lapply -> FUN -> colnames -> colnames
Execution halted
```

It looks like there is a problem when executing the R library. Do I need to fix something in my local R or is it a problem in the environment ?

Thanks

**Comment by steffenheyne** [comment]
it could be that this error is from histoneHMM in case there are no peaks found. At least I have no other idea. I think there is all fine with you R libs and env etc.

it seem you are using the predicted chip dict (Outputs/chip_seq_sample_config.PREDICTED.yaml) !? Can you make sure that in there is everything ok?
Eg. that you don't call peaks on an input sample. No input samples should have their own entry. They should just appear under the "control" key.

**Comment by Remi-Montagne** [comment]
My dictionary is like this:
```
chip_dict:
  SRX3823263_input:
    broad: true
    control: SRX3823275_control
```

 At the beginning the 2 samples were switched but I changed it manually to put the control in the "control" row. Should I switch them again ?

**Comment by steffenheyne** [comment]
but is SRX3823263_input an input/control sample as well? then skip the the "SRX3823263_input" entry completely. Otherwise it is fine. 
Do you have Macs2 broad peaks called? ie. did they run through without error for samples with `broad: True`? Just use these then.

**Comment by Remi-Montagne** [comment]
SRX3823263 is the input and SRX38275 the associated control. I wanted to keep it simple so it is the only 2 files I am working on right now.

So I must just remove this line ?

**Comment by steffenheyne** [comment]
in ChIP-seq the term "input" usually also refers to control samples. If SRX3823263 is your ChIP like H3K4me3 or a TF then your config is fine and leave it as it is.

**Comment by Remi-Montagne** [comment]
OK.

So the problem is that no peaks (or not enough) are found ?

**Comment by steffenheyne** [comment]
Probably. 

Can you try another sample, maybe even a "narrow" ChIP that you use with `broad: True` to see if histoneHMM can run properly in general for you?

And are there some broad peaks called by Macs2?

**Comment by Remi-Montagne** [comment]
So I tried with another dataset and had the same results.

However, I ran MACS2 only, outside of snakePipes, with a higher mfold ([2 50]) and this time the peak calling worked and returned peaks which seem in accordance with the ChIP-seq signal. Maybe my problem can be fixed only by changing MACS2 (and HistoneMM) parameters a little. 

I am not sure it can be done in command line ?

**Comment by mirax87** [comment]
I am facing the same issue when executing the snakePipes with broad peaks - I observed it in drosophila (dm6, ensembl91) and mouse (mm10, snakePipes' organism.yaml).
 
However, if I execute the command for the mouse samples using my local histoneHMM installation, the regions are produced (tried 2/4 samples) successfully.

Versions:
- local histoneHMM: from March 4, 2019 (according to git log).
- snakePipes 1.2.2

**Comment by katsikora** [comment]
this is now fixed in a)release 1.2.3 2)current develop branch
